crates/miden-objects/src/account/component/metadata/storage/schema.rs

```diff
@@ -0,0 +1,1202 @@
+    ) -> SchemaRequirementsIter<'_> {
+        if self.r#type == SchemaTypeIdentifier::void() || self.default_value.is_some() {
+            return Box::new(iter::empty());
+        }
+
+        let Some(name) = self.name.as_ref() else {
+            return Box::new(iter::empty());
+        };
+
+        Box::new(iter::once((
+            value_prefix.with_suffix(name),
+            InitValueRequirement {
+                description: self.description.clone(),
+                r#type: self.r#type.clone(),
+            },
+        )))
+    }
+
+    /// Validates that the defined felt type exists.
+    pub(crate) fn validate(&self) -> Result<(), AccountComponentTemplateError> {
+        let type_exists = SCHEMA_TYPE_REGISTRY.contains_felt_type(&self.felt_type());
+        if !type_exists {
+            return Err(AccountComponentTemplateError::InvalidType(
+                self.felt_type().to_string(),
+                "Felt".into(),
+            ));
+        }
+
+        if self.r#type == SchemaTypeIdentifier::void() {
+            if self.name.is_some() {
+                return Err(AccountComponentTemplateError::InvalidSchema(
+                    "void felt elements must be unnamed".into(),
+                ));
+            }
+            if self.default_value.is_some() {
+                return Err(AccountComponentTemplateError::InvalidSchema(
+                    "void felt elements cannot define `default-value`".into(),
+                ));
+            }
+            return Ok(());
+        }
+
+        if self.name.is_none() {
+            return Err(AccountComponentTemplateError::InvalidSchema(
+                "non-void felt elements must be named".into(),
+            ));
+        }
+
+        if let Some(value) = self.default_value {
+            validate_felt_value(&self.felt_type(), value)
+                .map_err(AccountComponentTemplateError::StorageValueParsingError)?;
+        }
+        Ok(())
+    }
+}
+
+impl Serializable for FeltSchema {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write(&self.name);
+        target.write(&self.description);
+        target.write(&self.r#type);
+        target.write(self.default_value);
+    }
+}
+
+impl Deserializable for FeltSchema {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let name = Option::<StorageValueName>::read_from(source)?;
+        let description = Option::<String>::read_from(source)?;
+        let r#type = SchemaTypeIdentifier::read_from(source)?;
+        let default_value = Option::<Felt>::read_from(source)?;
+        Ok(FeltSchema { name, description, r#type, default_value })
+    }
+}
+
+// VALUE VALIDATION HELPERS
+// ================================================================================================
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum WordTypeKind {
+    Word,
+    Felt,
+}
+
+fn word_type_kind(schema_type: &SchemaTypeIdentifier) -> WordTypeKind {
```

Comment:
Nit: Could this function be a method on `SchemaTypeIdentifier` instead?

Reply:
I'm thinking this is more type registry-related. If the registry is somehow user-extensible, it will become the source of truth for identifying types and what they map to in terms of converting to native types. I'll note this in the list of followups